### PR TITLE
fix: add ellipsis for permit fiat values

### PR DESCRIPTION
**Pull request by OGPoyraz**
## **Description**

This PR goal is to add ellipsis to the fiat value if it's more than 15 character in permit simulations.

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/26001?quickstart=1)

## **Related issues**

Fixes: https://github.com/MetaMask/MetaMask-planning/issues/2842

Follow-up Cherry-picked PR: 
 https://github.com/MetaMask/metamask-extension/pull/26515

## **Manual testing steps**

1. Go to cowswap
2. Swap a token with gas-free approve for another token
3. Notice the permit signature screen

### **After**

Screenshot 2024-07-22 at 14 40 57

## **Pre-merge author checklist**

- [X] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [X] I've completed the PR template to the best of my ability
- [X] I’ve included tests if applicable
- [X] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [X] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Review by vinistevam — DISMISSED** [review]

**Comment by vinistevam** [comment]
Great PR @OGPoyraz ! Just a small lint error (jsdocs) failing the pipeline.

**Review by jpuri — DISMISSED** [review]

File: ui/hooks/useFiatFormatter.ts [hunk 1]
```diff
@@ -1,35 +1,89 @@
...
  * @returns A function that takes a fiat amount as a number and returns a formatted string.
  */
 
+const TRUNCATED_CHAR_LIMIT_FOR_SHORTENED_FIAT = 15;
+const TRUNCATED_START_CHARS_FOR_SHORTENED_FIAT = 12;
+
 type FiatFormatter = (fiatAmount: number) => string;
 
-export const useFiatFormatter = (): FiatFormatter => {
+export const useFiatFormatter = (shortenFiatValue?: boolean): FiatFormatter => {
```
Comment (matthewwalsh0) [inline ui/hooks/useFiatFormatter.ts:28]: Could this instead go on the returned function itself so the same hook can be re-used in either format?

Comment (matthewwalsh0) [inline ui/hooks/useFiatFormatter.ts:28]: Since this is a boolean, should we use an object argument to make it more readable?

Comment (matthewwalsh0) [inline ui/hooks/useFiatFormatter.ts:28]: Minor, but could we just call this `shorten` since fiat value is implied?

**Review by matthewwalsh0 — CHANGES_REQUESTED** [review]

**Review by matthewwalsh0 — DISMISSED** [review]

**Review by jpuri — APPROVED** [review]

**Review by matthewwalsh0 — APPROVED** [review]

**Comment by digiwand** [comment]
Follow-up PR adds tooltip when ellipsis is shown and updates calculation with decimals
https://github.com/MetaMask/metamask-extension/pull/26523

**Comment by digiwand** [comment]
Cherry-pick PR into 12.1.0 https://github.com/MetaMask/metamask-extension/pull/26515

**Pull request merged**